### PR TITLE
[SPARK-23238][SQL] Externalize SQLConf configurations exposed in documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to expose few internal configurations found in the documentation.

Also it fixes the description for `spark.sql.execution.arrow.enabled`. 
It's quite self-explanatory.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by ueshin** [comment]
LGTM.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by ueshin** [comment]
Jenkins, retest this please.

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1045,9 +1045,10 @@ object SQLConf {
     buildConf("spark.sql.execution.arrow.enabled")
       .internal()
       .doc("Make use of Apache Arrow for columnar data transfers. Currently available " +
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `When true, make use of`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -1045,9 +1045,10 @@ object SQLConf {
     buildConf("spark.sql.execution.arrow.enabled")
       .internal()
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: This conf will be mentioned in the doc `https://github.com/apache/spark/pull/19575`.  I think this is not qualified as an internal conf.

Comment (gatorsmile) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Could you change it in this PR and use this JIRA number https://issues.apache.org/jira/browse/SPARK-23238

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Yup.

**Review by BryanCutler — APPROVED** [review]
LGTM

**Comment by gatorsmile** [comment]
LGTM

**Comment by BryanCutler** [comment]
this has been failing due to #19892 which was recently reverted

**Comment by BryanCutler** [comment]
jenkins retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Review by felixcheung — APPROVED** [review]

**Comment by viirya** [comment]
LGTM

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
ok to test

**Comment by viirya** [comment]
retest this please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -1043,11 +1043,11 @@ object SQLConf {
...
-        "for use with pyspark.sql.DataFrame.toPandas with the following data types: " +
-        "StringType, BinaryType, BooleanType, DoubleType, FloatType, ByteType, IntegerType, " +
-        "LongType, ShortType")
+      .doc("When true, make use of Apache Arrow for columnar data transfers. Currently available " +
+        "for use with pyspark.sql.DataFrame.toPandas, and " +
+        "pyspark.sql.SparkSession.createDataFrame when its input is a Pandas DataFrame. " +
+        "The following data types are unsupported: " +
+        "MapType, ArrayType of TimestampType, and nested StructType.")
       .booleanConf
       .createWithDefault(false)
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `spark.sql.execution.arrow.maxRecordsPerBatch` is also mentioned in the doc change at #19575. Shall we also externalize it?

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Yup. Let me

**Comment by viirya** [comment]
Let's also update the PR title.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 4]
```diff
@@ -1043,17 +1041,16 @@ object SQLConf {
...
 
   val ARROW_EXECUTION_MAX_RECORDS_PER_BATCH =
     buildConf("spark.sql.execution.arrow.maxRecordsPerBatch")
-      .internal()
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I am not sure about conf. https://github.com/apache/spark/pull/19575/files#r164252424 

If we want to merge this PR now, maybe revert this change?

Comment (gatorsmile) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: We can externalize this conf in that PR https://github.com/apache/spark/pull/19575, if we believe this conf is the one we will use in the long term.

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Makes sense. Let me take this out. Is this only one you are concerned of for now?

**Comment by gatorsmile** [comment]
LGTM

**Review by viirya — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.

Thanks @ueshin, @gatorsmile, @BryanCutler, @felixcheung and @viirya.
